lang/cs/Org.Apache.REEF.Network/Group/Task/Impl/OperatorTopology.cs

```diff
@@ -352,21 +355,45 @@ private void ScatterHelper(IList<T> messages, IList<NodeStruct<T>> order, int nu
         /// Checks if the identifier is registered with the Name Server.
         /// Throws exception if the operation fails more than the retry count.
         /// </summary>
-        /// <param name="identifier">The identifier to look up</param>
-        /// <param name="retries">The number of times to retry the lookup operation</param>
-        private void WaitForTaskRegistration(string identifier, int retries)
+        /// <param name="identifiers">The identifier to look up</param>
+        private void WaitForTaskRegistration(IList<string> identifiers)
         {
-            for (int i = 0; i < retries; i++)
+            using (Logger.LogFunction("OperatorTopology::WaitForTaskRegistration"))
             {
-                if (_nameClient.Lookup(identifier) != null)
+                IList<string> foundList = new List<string>();
+                try
+                {
+                    for (var i = 0; i < _retryCount; i++)
```

Comment:
Yes, one retrycount per connection. Wouldn't it be natural to try each connection n times, instead of tolerating n failures to connect overall?